js/site-nav.js

```diff
@@ -47,21 +49,33 @@ SiteNav.prototype.hide = function() {
   this.isOpen = false;
 };
 
-SiteNav.prototype.handleChange = function(e) {
-  var $subToggle = $(e.currentTarget);
-  var checked = $subToggle.is(':checked');
-  var $list = $(e.currentTarget).siblings('ul');
+SiteNav.prototype.toggleSublist = function(e) {
+  var method = this.$openSublist ? this.hideSublist : this.showSublist;
+  var $target = $(e.target);
+  var $sublistParent = $target.hasClass('js-sublist-parent') ? $target : $target.closest('.js-sublist-parent');
```

Comment:
Since `closest` tests the target element as well as its parents, we should be able to simplify this as `var $sublistParent = $target.closest('.js-sublist-parent')`:

https://api.jquery.com/closest/